File: datamodel/src/main/java/datamodel/clinical/CypInteraction.java
package datamodel.clinical;

import org.immutables.value.Value;

@Value.Immutable
public interface CypInteraction {

    enum Type {
        INDUCER,
        INHIBITOR,
        SUBSTRATE
    }

    enum Strength {
        NONE,
        STRONG,
        MODERATE,
        WEAK,
        SENSITIVE,
        MODERATE_SENSITIVE
    }

    Type type();

    Strength strength();

    String cyp();
}

File: algo/src/main/kotlin/com/hartwig/actin/datamodel/clinical/CypInteraction.kt
package com.hartwig.actin.datamodel.clinical

data class CypInteraction(
    val type: Type,
    val strength: Strength,
    val cyp: String,
) {

    enum class Type {
        INDUCER,
        INHIBITOR,
        SUBSTRATE
    }

    enum class Strength {
        STRONG,
        MODERATE,
        WEAK,
        SENSITIVE,
        MODERATE_SENSITIVE
    }
}
